### PR TITLE
feat: `getBalanceChanges`

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR implements `getBalanceChanges`, formerly known as `parseBalanceChanges` in `ripple-lib-transactionparser`. This was used when a transaction's result was returned in 1.x.

### Context of Change

Makes it easier to parse the results of a transaction. Requested by @mDuo13 

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Test Plan

CI passes.

### CODE REVIEW

**Review by mDuo13 — APPROVED** [review]
Pointed out some things where I'm not sure if the TypeScript definitions will work, but otherwise looks good to me.

Confirmed it works smoothly in the "Send XRP" tutorial as well. Code:

    const tx = await api.request({
      command: "tx",
      transaction: txID,
      min_ledger: earliestLedgerVersion,
      max_ledger: lastLedgerSequence
    })

    block.find(".output-area").html(
      `<div><strong>Transaction result code:</strong>
      ${tx.result.meta.TransactionResult} (${tx.result.validated ? "validated": "pending"})</div>
      <div><strong>Balance changes:</strong>
      <pre><code>${pretty_print(xrpl.getBalanceChanges(tx.result.meta))}</code></pre>
      </div>`
    )

And result:

Screenshot from 2021-10-05 15-46-16

File: src/models/transactions/metadata.ts [hunk 1]
```diff
@@ -27,11 +27,10 @@ interface DeletedNode {
   }
 }
 
-type Node = CreatedNode | ModifiedNode | DeletedNode
+export type Node = CreatedNode | ModifiedNode | DeletedNode
 
 export default interface TransactionMetadata {
   AffectedNodes: Node[]
-  DeliveredAmount?: Amount
```
[the remark this answers is not in the document]
Comment (mvadari) [reply src/models/transactions/metadata.ts:34]: Ah okay, I thought it was a typo or something.

File: src/models/transactions/metadata.ts [hunk 2]
```diff
@@ -27,11 +27,10 @@ interface DeletedNode {
...
   delivered_amount?: Amount
```
[the remark this answers is not in the document]
Comment (mvadari) [reply src/models/transactions/metadata.ts:34]: It's fine - `Amount` is an `IssuedCurrencyAmount` or a `string` (to handle XRP amounts). I explicitly added it in while I was here though.

File: test/utils/getBalanceChanges.ts [hunk 3]
```diff
@@ -0,0 +1,454 @@
+import { assert } from 'chai'
+
+import { getBalanceChanges } from '../../src/utils'
+import paymentIou from '../fixtures/utils/paymentIou.json'
```
[the remark this answers is not in the document]
Comment (mvadari) [reply test/utils/getBalanceChanges.ts:4]: Will do. These testcases were pulled straight from ripple-lib-transactionparser so I didn't put a lot of thought into changing it.

**Review by natenichols — APPROVED** [review]

File: src/utils/balanceChanges.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  node: NormalizedNode,
+): { account: string; balance: Balance } | null {
+  const value = computeBalanceChange(node)
+
+  if (value === null) {
+    return null
+  }
+
+  return {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- okay here
```
Comment (natenichols) [inline src/utils/balanceChanges.ts:107]: Couldn't this still be undefined, though, if both `Accounts` are undefined?

**Comment by natenichols** [comment]
Left a few comments, nothing worth holding up approval

File: src/utils/balanceChanges.ts (hunk 4 above)
Comment (mvadari) [reply src/utils/balanceChanges.ts:107]: My understanding of these meta fields is that one of them has to be defined. I might be wrong though, can @mDuo13 confirm?

**Pull request merged**